testplan/web_ui/testing/src/Toolbar/Toolbar.js

```diff
@@ -57,409 +65,330 @@ library.add(
   faAngleDown
 );
 
-/**
- * Toolbar component, contains the toolbar buttons & Filter box.
- */
-class Toolbar extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      treeView: true,
-      displayTime: false,
-      displayPath: false,
-      filterOpen: false,
-      infoModal: false,
-      filter: "all",
-      displayEmpty: true,
-      displayTags: false,
-      expandStatus: this.props.expandStatus,
-    };
-
-    this.filterOnClick = this.filterOnClick.bind(this);
-    this.toggleInfoOnClick = this.toggleInfoOnClick.bind(this);
-    this.toggleTreeView = this.toggleTreeView.bind(this);
-    this.toggleTimeDisplay = this.toggleTimeDisplay.bind(this);
-    this.togglePathDisplay = this.togglePathDisplay.bind(this);
-    this.toggleEmptyDisplay = this.toggleEmptyDisplay.bind(this);
-    this.toggleTagsDisplay = this.toggleTagsDisplay.bind(this);
-    this.toggleFilterOnClick = this.toggleFilterOnClick.bind(this);
-    this.toggleExpand = this.toggleExpand.bind(this);
-    this.toggleCollapse = this.toggleCollapse.bind(this);
-  }
-
-  toggleTreeView() {
-    this.props.updateTreeViewFunc(!this.state.treeView);
-    this.setState((prevState) => ({
-      treeView: !prevState.treeView,
-    }));
-  }
-
-  togglePathDisplay() {
-    this.props.updatePathDisplayFunc(!this.state.displayPath);
-    this.setState((prevState) => ({
-      displayPath: !prevState.displayPath,
-    }));
-  }
-
-  toggleTimeDisplay() {
-    this.props.updateTimeDisplayFunc(!this.state.displayTime);
-    this.setState((prevState) => ({
-      displayTime: !prevState.displayTime,
-    }));
-  }
-
-  toggleInfoOnClick() {
-    this.setState((prevState) => ({
-      infoModal: !prevState.infoModal,
-    }));
-  }
-
-  toggleFilterOnClick() {
-    this.setState((prevState) => ({
-      filterOpen: !prevState.filterOpen,
-    }));
-  }
-
-  filterOnClick(e) {
-    let checkedValue = e.currentTarget.value;
-    this.setState({ filter: checkedValue });
-    this.props.updateFilterFunc(checkedValue);
-  }
-
-  toggleEmptyDisplay() {
-    this.props.updateEmptyDisplayFunc(!this.state.displayEmpty);
-    this.setState((prevState) => ({
-      displayEmpty: !prevState.displayEmpty,
-    }));
-  }
-
-  toggleTagsDisplay() {
-    this.props.updateTagsDisplayFunc(!this.state.displayTags);
-    this.setState((prevState) => ({
-      displayTags: !prevState.displayTags,
-    }));
-  }
-
-  toggleExpand() {
-    if (this.props.expandStatus === EXPAND_STATUS.EXPAND) {
-      this.props.updateExpandStatusFunc(EXPAND_STATUS.DEFAULT);
+const ToolbarExpandButtons = ({
```

Comment:
You dropped the original docstring.

Reply:
None of the button functions had docstrings, and they have talking names. The Toolbar component still has it's original docstring.

Reply:
None of the button functions had docstrings, and they have talking names. The Toolbar component still has it's original docstring.

Reply:
Up to you.